apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/ItemService.java

```diff
@@ -200,6 +205,26 @@ public Item update(Item item) {
 
   private boolean checkItemValueLength(long namespaceId, String value) {
     int limit = getItemValueLengthLimit(namespaceId);
+    Namespace currentNamespace = namespaceService.findOne(namespaceId);
+    if(currentNamespace != null) {
+      String appId = currentNamespace.getAppId();
+      String namespaceName = currentNamespace.getNamespaceName();
+      String clusterName = currentNamespace.getClusterName();
+      if(appId != null && namespaceName != null && clusterName != null) {
+        Matcher m = clusterPattern.matcher(clusterName);
+        boolean isGray = m.matches();
+        if (isGray) {
+          Cluster currentCluster = clusterService.findOne(appId, clusterName);
+          long parentClusterId = currentCluster.getParentClusterId();
+          Cluster parentCluster = clusterService.findOne(parentClusterId);
+          Namespace namespace = namespaceService.findOne(appId, parentCluster.getName(), namespaceName);
```

Comment:
Why not use `NamespaceService#findParentNamespace(com.ctrip.framework.apollo.biz.entity.Namespace)` instead?